Storage/Bookmarks/BookmarksModel.swift

```diff
@@ -106,6 +110,14 @@ public class BookmarksModel: BookmarksModelFactorySource {
     }
 
     /**
+     * Produce a new model with a memory-backed root with the given GUID removed from the current folder
+     */
+    public func removeGUIDFromCurrent(guid: GUID) -> BookmarksModel {
+        let memoryBackedFolder = MemoryBookmarkFolder.fromBookmarkFolder(self.current)
```

Comment:
This is kinda backwards. You make a copy of the current folder then mutate it! Instead, have `removeItemWithGUID` return a new `MemoryBookmarkFolder`, like this:

```
    override public func removeItemWithGUID(guid: GUID) -> MemoryBookmarkFolder {
        let without = self.children.filter({$0.guid != guid})
        return MemoryBookmarkFolder(guid: self.guid, title: self.title, children: without)
    }
```

then this can just be:

```
    public func removeGUIDFromCurrent(guid: GUID) -> BookmarksModel {
        return BookmarksModel(modelFactory: self.factory, root: self.current.removeItemWithGUID(guid))
    }
```